src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -972,9 +959,14 @@ int Terminal::_VisibleEndIndex() const noexcept
 
 Viewport Terminal::_GetVisibleViewport() const noexcept
 {
+    // GH#3493: if we're in the alt buffer, then it's possible that the mutable
+    // viewport's size hasn't been updated yet. In that case, use the
+    // temporarily stashed _altBufferSize instead.
     const COORD origin{ 0, gsl::narrow<short>(_VisibleStartIndex()) };
+    const COORD size{ _inAltBuffer() ? _altBufferSize.to_win32_coord() :

```

Comment:
This is almost aching for another helper ... or, seems like you could use the Dimensions off the GetMutableViewport helper's return value? I'd prefer concentrating the magic in one place. :)

Reply:
Can't we make a `til::rect _viewports[2]` and index into it with a `bool _altBufferInUse`?
Like... `_viewports[_altBufferInUse].to_win32_coord()` or something like that...

Reply:
Bear in mind that we'll probably need to support more buffers than just main and alt in the long term. So I'd expect properties like this would ultimately be part of the text buffer itself, and then you could just do something like `_activeBuffer().mutableViewport`. But we also need to get the architecture synced up with conhost, which currently manages this sort of state in the `SCREEN_INFORMATION` class. Not that I'm saying this is essential for now, but if you're trying to clean up the code, it's best to also consider how we are likely to extend this in the future.